core/server/api/authentication.js

```diff
@@ -260,22 +273,21 @@ authentication = {
         });
     },
 
-    revoke: function (object) {
-        var token;
-
-        if (object.token_type_hint && object.token_type_hint === 'access_token') {
-            token = dataProvider.Accesstoken;
-        } else if (object.token_type_hint && object.token_type_hint === 'refresh_token') {
-            token = dataProvider.Refreshtoken;
-        } else {
-            return errors.BadRequestError('Invalid token_type_hint given.');
+    updateSetup: function updateSetup(object, options) {
+        if (!options.context || !options.context.user) {
+            return Promise.reject(new errors.NoPermissionError('You are not logged in.'));
         }
 
-        return token.destroyByToken({token: object.token}).then(function () {
-            return Promise.resolve({token: object.token});
-        }, function () {
-            // On error we still want a 200. See https://tools.ietf.org/html/rfc7009#page-5
-            return Promise.resolve({token: object.token, error: 'Invalid token provided'});
+        return users.read({id: options.context.user, include: 'roles'}).then(function (result) {
```

Comment:
This line is the same as `return dataProvider.User.findOne({role: 'Owner', status: 'all'});` - doing it via the dataProvider (model) would mean not needing to include users. Sorry if I confused you with this.